packages/phoenix-event-display/src/tests/managers/ui-manager/phoenix-menu/phoenix-menu-ui.test.ts

```diff
@@ -0,0 +1,124 @@
+import { ThreeManager } from '../../../../managers/three-manager';
+import { PhoenixMenuNode } from '../../../../managers/ui-manager/phoenix-menu/phoenix-menu-node';
+import { PhoenixMenuUI } from '../../../../managers/ui-manager/phoenix-menu/phoenix-menu-ui';
+
+jest.mock('../../../../managers/three-manager');
+
+describe('PhoenixMenuUI', () => {
+  let phoenixMenuUI: PhoenixMenuUI;
+  let phoenixMenuUIPrivate: any;
+  let phoenixMenuRoot: PhoenixMenuNode;
+  let three: ThreeManager;
+
+  beforeEach(() => {
+    three = new ThreeManager(new InfoLogger());
+    phoenixMenuRoot = new PhoenixMenuNode('test');
+    phoenixMenuUI = new PhoenixMenuUI(phoenixMenuRoot, three);
+    phoenixMenuUIPrivate = phoenixMenuUI as any;
+  });
+
+  afterEach(() => {
+    phoenixMenuUI.clear();
+  });
+
+  it('should create an instance', () => {
+    expect(phoenixMenuUI).toBeTruthy();
+  });
+
+  it('should add geometry (detector geometry) folder to the menu', () => {
+    phoenixMenuUI.addGeometryFolder();
+
+    expect(phoenixMenuUIPrivate.geomFolder.name).toBe('Detector');
+    expect(phoenixMenuUIPrivate.geomFolder).toBeInstanceOf(PhoenixMenuNode);
+    expect(phoenixMenuUIPrivate.geomFolder.toggleState).toBeTruthy();
+    expect(phoenixMenuUIPrivate.geomFolder.configs[0].type).toBe('checkbox');
+    expect(phoenixMenuUIPrivate.geomFolder.configs[1].type).toBe('slider');
+  });
+
+  it('should add event data folder with functions for event data toggles like show/hide and depthTest', () => {
+    phoenixMenuUI.addEventDataFolder();
+
+    expect(phoenixMenuUIPrivate.eventFolder.name).toBe('Event Data');
+    expect(phoenixMenuUIPrivate.eventFolder).toBeInstanceOf(PhoenixMenuNode);
+    expect(phoenixMenuUIPrivate.eventFolder.toggleState).toBeTruthy();
+    expect(phoenixMenuUIPrivate.eventFolder.configs[0].type).toBe('checkbox');
+    expect(phoenixMenuUIPrivate.eventFolder.childrenActive).toBeFalsy();
+    expect(phoenixMenuUIPrivate.eventFolder.parent).toBeInstanceOf(
+      PhoenixMenuNode
+    );
+    expect(phoenixMenuUIPrivate.eventFolder.parent.name).toBe('test');
+  });
+
+  it('should add labels folder to the menu', () => {
+    phoenixMenuUI.addLabelsFolder('test');
+
+    expect(phoenixMenuUIPrivate.labelsFolder.name).toBe('Labels');
+    expect(phoenixMenuUIPrivate.labelsFolder).toBeInstanceOf(PhoenixMenuNode);
+    expect(phoenixMenuUIPrivate.labelsFolder.toggleState).toBeTruthy();
+    expect(phoenixMenuUIPrivate.labelsFolder.configs[0].type).toBe('slider');
+    expect(phoenixMenuUIPrivate.labelsFolder.configs[1].type).toBe('color');
+    expect(phoenixMenuUIPrivate.labelsFolder.configs[2].type).toBe('button');
+    expect(phoenixMenuUIPrivate.labelsFolder.parent).toBeInstanceOf(
+      PhoenixMenuNode
+    );
+    expect(phoenixMenuUIPrivate.labelsFolder.parent.name).toBe('test');
+  });
+
+  it('should remove label from the menu and scene if it exists', () => {
+    phoenixMenuUI.addLabelsFolder('test');
+    phoenixMenuUI.removeLabel('test');
```

Comment:
This does not test the intended behavior. Pass in a mock label node of type `PhoenixMenuNode` and make sure that it's `remove` method was called.

Reply:
Done ✅. Thanks!